### DISCUSSION TITLE
SetChildNeedsDisplay();  .SetNeedsDisplay();

### DISCUSSION THREAD

**Discussion by GronbergInfo** [tznind/TerminalGuiDesigner#216] [General]
Hi. i hav ejust started noticing that listview does not update with new lines.
I add a list<string> as Setsource to a listview, adding a bunch of rows into that list, only time the Listview updates/redraw is when i actually click on it in the console.

I have tried to add a bunch of lines and then run both SetChildNeedsDisplay();  .SetNeedsDisplay(); for my listview, and it still only updates the new lines when i click the object it self.

Is there a setting or something for the whole view that i can run re-draw/update everything?

**Comment by BDisp** [comment]
That is a old issue when a view is been updating from a thread it's needed to call the main thread with `Application.Mainloop.Invoke(() => myLV.SetNeedsDisplay());`. You are only setting the `SetNeedsDisplay` and when you click on the console the mainloop runs and update the `ListView`.

**Reply by GronbergInfo** [reply]
Hi.
As a test, i added "Application.MainLoop.Invoke(() => )" on basically anything that touches the UI, but i still dont get any UI update until i click something in the console.

Take this an one of the example that does not show until i click on the main window letting the main thread run:
```
 if (!port.IsOpen)
            {
                port = new SerialPort(settings.ServerSettings.ComSelection, settings.ServerSettings.BaudRate, settings.ServerSettings.ComParity, settings.ServerSettings.Datasize, settings.ServerSettings.StopBits);
                Application.MainLoop.Invoke(() => lstServerLogsList.Add("COM Server settings:"));
                Application.MainLoop.Invoke(() => lstServerLogsList.Add("COM Port: " + settings.ServerSettings.ComSelection));
                Application.MainLoop.Invoke(() => lstServerLogsList.Add("Baud rate: " + settings.ServerSettings.BaudRate));
                Application.MainLoop.Invoke(() => lstServerLogsList.Add("Parity: " + settings.ServerSettings.ComParity));
                Application.MainLoop.Invoke(() => lstServerLogsList.Add("Data Size: " + settings.ServerSettings.Datasize));
                Application.MainLoop.Invoke(() => lstServerLogsList.Add("Stop Bits: " + settings.ServerSettings.StopBits));
                Application.MainLoop.Invoke(() => lstServerLogsList.Add("---"));


                // Attach a method to be called when there
                // is data waiting in the port's buffer
                Application.MainLoop.Invoke(() => lstServerLogsList.Add("Incoming Data:"));
                port.DataReceived += new SerialDataReceivedEventHandler(port_DataReceived);
                // Open the port for communications
                port.Open();

            }
```

**Reply by BDisp** [reply]
You are forgetting the most essential that is call the `SetNeedsDisplay()`. You don't need call `Application.MainLoop.Invoke` for each added items in the list. You only need use it at end to call the `SetNeedsDisplay()` from your `ListView`.
